src/wp-includes/blocks/file.php

```diff
@@ -71,19 +54,37 @@ static function ( $matches ) {
 	);
 
 	// If it uses the Interactivity API, add the directives.
-	if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN && $should_load_view_script ) {
+	if ( $should_load_view_script ) {
 		$processor = new WP_HTML_Tag_Processor( $content );
 		$processor->next_tag();
 		$processor->set_attribute( 'data-wp-interactive', '' );
 		$processor->next_tag( 'object' );
-		$processor->set_attribute( 'data-wp-bind--hidden', '!selectors.core.file.hasPdfPreview' );
-		$processor->set_attribute( 'hidden', true );
+		$processor->set_attribute( 'data-wp-style--display', 'selectors.core.file.hasPdfPreview' );
 		return $processor->get_updated_html();
 	}
 
 	return $content;
 }
 
+/**
+ * Ensure that the view script has the `wp-interactivity` dependency.
+ *
+ * @since 6.4.0
+ *
+ * @global WP_Scripts $wp_scripts
+ */
+function block_core_file_ensure_interactivity_dependency() {
+	global $wp_scripts;
+	if (
+		isset( $wp_scripts->registered['wp-block-file-view'] ) &&
+		! in_array( 'wp-interactivity', $wp_scripts->registered['wp-block-file-view']->deps, true )
+	) {
+		$wp_scripts->registered['wp-block-file-view']->deps[] = 'wp-interactivity';
+	}
+}
+
+add_action( 'wp_print_scripts', 'block_core_file_ensure_interactivity_dependency' );
```

Comment:
This approach is a workaround that results in duplicated code and can lead to problems with the dependency tree. The scripts should be registered with that dependency from the beginning.

It looks like the `core/file` block and a few other blocks that currently require the `wp-interactivity` script as a dependency also have `supports.interactivity = true` in their block metadata. It would be more appropriate to rely on that, in order to provide a generally applicable solution. For example:
* Update the `register_block_script_handle()` function.
* If the block metadata includes interactivity support, register the block view script with the `wp-interactivity` script as a dependency.